File: 250424/운행 되고 있는 시간/hours-in-service.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");

const n = Number(input[0]);
const segments = input.slice(1, n + 1).map(line => line.split(' ').map(Number));

let max = Number.MIN_SAFE_INTEGER
for(let i = 0; i< segments.length; i++){
    const visited = Array(1001).fill(false)
    const temp = [...segments.slice(0, i), ...segments.slice(i + 1, segments.length)];
    for(const [start, end] of temp){
        for(let j = start; j < end; j++){
            visited[j] = true
        }
    }
    max = Math.max(max, visited.reduce((prev, cur) => prev + (cur ? 1 : 0)));
}

console.log(max)

File: 250810/운행 되고 있는 시간/hours-in-service.py
n = int(input())
times = [tuple(map(int, input().split())) for _ in range(n)]
a = [t[0] for t in times]
b = [t[1] for t in times]

max_count = 0
for i in range(n):                
    temp = [False] * 1001
    for k in range(n):
        if i == k:
            continue
        for t in range(a[k], b[k]):
            temp[t] = True
    max_count = max(max_count, temp.count(True))

print(max_count)